internal/provider/resource_aws_transit_gateway_attachment.go

```diff
@@ -0,0 +1,319 @@
+package provider
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-network/preview/2020-09-07/client/network_service"
+	networkmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-network/preview/2020-09-07/models"
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+var tgwDefaultTimeout = time.Minute * 1
+var tgwCreateTimeout = time.Minute * 35
+var tgwDeleteTimeout = time.Minute * 35
+
+func resourceAwsTransitGatewayAttachment() *schema.Resource {
+	return &schema.Resource{
+		Description: "The AWS Transit gateway attachment resource allows you to manage a Transit gateway attachment that attaches an HVN to a Transit gateway in AWS.",
+
+		CreateContext: resourceAwsTransitGatewayAttachmentCreate,
+		ReadContext:   resourceAwsTransitGatewayAttachmentRead,
+		DeleteContext: resourceAwsTransitGatewayAttachmentDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Default: &tgwDefaultTimeout,
+			Create:  &tgwCreateTimeout,
+			Delete:  &tgwDeleteTimeout,
+		},
+		// Importer: &schema.ResourceImporter{
+		// 	State: schema.ImportStatePassthrough,
+		// },
+
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"hvn_id": {
+				Description:      "The ID of the HashiCorp Virtual Network (HVN).",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			"transit_gateway_attachment_id": {
+				Description:      "The ID of the Transit gateway attachment.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			"transit_gateway_id": {
+				Description: "The ID of the Transit gateway in AWS.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"resource_share_arn": {
+				Description: "The Amazon Resource Name (ARN) of the Resource Share that is needed to grant HCP access to the Transit gateway in AWS.",
+				Type:        schema.TypeString,
+				Required:    true,
+				Sensitive:   true,
+				ForceNew:    true,
+			},
+			"destination_cidrs": {
```

Comment:
Nit: Our internal implementation used the `destination_cidrs` name that was copied here, but I haven't seen the "destination" qualifier anywhere else. Would `cidrs` be more concise/clear?

Reply:
I like the concise `cidrs`, assuming the AWS provider uses the same. Though I'd defer to @xargs-P

Reply:
this is essentially the same what AWS calls `destination-cidr-block` https://docs.aws.amazon.com/cli/latest/reference/ec2/create-route.html. Relatively soon we will introduce a concept of HVN Routes that would have destination-cidr (ip range) and destination (peering, tgw attachment, etc). I guess we can call it just `cidr`. In the peering, for example, it's called `peer_vpc_cidr_block`.

Reply:
Given @smaant 's response, I'm going to leave this as `destination_cidrs`. Also note that the UI field is labeled "Destinations", so I think this aligns well with that.